lib/src/importer/node_package.dart

```diff
@@ -0,0 +1,394 @@
+// Copyright 2023 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:collection/collection.dart';
+import 'package:sass/src/util/map.dart';
+import 'package:sass/src/util/nullable.dart';
+
+import '../importer.dart';
+import './utils.dart';
+import 'dart:convert';
+import '../io.dart';
+import 'package:path/path.dart' as p;
+
+/// An [Importer] that resolves `pkg:` URLs using the Node resolution algorithm.
+class NodePackageImporterInternal extends Importer {
+  final Uri entryPointURL;
```

Comment:
* Please document this.
* This should be private.
* [Capitalize acronyms and abbreviations longer than two letters like words](https://dart.dev/effective-dart/style#do-capitalize-acronyms-and-abbreviations-longer-than-two-letters-like-words)